internal/services/network/application_gateway_resource.go

```diff
@@ -4760,7 +4760,9 @@ func checkBasicSkuFeatures(d *pluginsdk.ResourceDiff) error {
 func applicationGatewayCustomizeDiff(ctx context.Context, d *pluginsdk.ResourceDiff, _ interface{}) error {
 	_, hasAutoscaleConfig := d.GetOk("autoscale_configuration.0")
 	capacity, hasCapacity := d.GetOk("sku.0.capacity")
-	tier := d.Get("sku.0.tier").(string)
+	oldTierInterface, newTierInterface := d.GetChange("sku.0.tier")
```

Comment:
Please rename to `oldSkuTier` and `newSkuTier`. The type of the variable is already apparent, and not relevant to be included in the name.